### ISSUE TITLE
[master] iostreams deprecation warnings in boost 1.61.

### ISSUE THREAD

**Issue by vikasnkumar** [libbitcoin/libbitcoin#582]
While compiling `libbitcoin` with `boost 1.61` on Kali Linux 2.0 Rolling release, I get the following warnings from the bitcoin header files. There are several 100 of these showing up when compiling `libbitcoin-explorer` too.

```
In file included from /usr/include/boost/iostreams/detail/is_dereferenceable.hpp:13:0,
                 from /usr/include/boost/iostreams/detail/resolve.hpp:26,
                 from /usr/include/boost/iostreams/detail/push.hpp:24,
                 from /usr/include/boost/iostreams/detail/streambuf/indirect_streambuf.hpp:31,
                 from /usr/include/boost/iostreams/stream_buffer.hpp:22,
                 from /usr/include/boost/iostreams/stream.hpp:21,
                 from /usr/local/include/bitcoin/bitcoin/utility/container_sink.hpp:25,
/usr/include/boost/type_traits/detail/template_arity_spec.hpp:13:84: note: #pragma message: NOTE: Use of this header (template_arity_spec.hpp) is deprecated
 # pragma message("NOTE: Use of this header (template_arity_spec.hpp) is deprecated")
```

**Comment by evoskuil** [comment]
Thanks! We'll see if we can't drop use of this header before moving to suppress.

**Comment by evoskuil** [comment]
Actually it looks like this just comes with iostreams, so we'd have to suppress it - but #pragma messages [cannot be suppressed](http://stackoverflow.com/questions/35328709/disable-g-pragma-message). This is a [boost bug](https://svn.boost.org/trac/boost/ticket/11860) that is fixed in 1.62.
